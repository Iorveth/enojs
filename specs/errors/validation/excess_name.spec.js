const eno = require('../../../eno.js');

const input = 'language: eno';

describe('validation.excessName', () => {
  describe('without a custom message', () => {
    const document = eno.parse(input);

    let error;
    try {
      document.assertAllTouched();
    } catch(err) {
      error = err;
    }

    it(`provides a correct message`, () => {
      expect(error.message).toMatchSnapshot();
    });

    it(`provides correct selection metadata`, () => {
      expect(error.selection).toMatchSnapshot();
    });
  });

  describe('with a custom message', () => {
    const document = eno.parse(input);

    let error;
    try {
      document.assertAllTouched('my custom message');
    } catch(err) {
      error = err;
    }

    it(`provides a correct message`, () => {
      expect(error.message).toMatchSnapshot();
    });

    it(`provides correct selection metadata`, () => {
      expect(error.selection).toMatchSnapshot();
    });
  });
});
