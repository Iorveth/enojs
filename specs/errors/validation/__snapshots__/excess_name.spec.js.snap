// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation.excessName with a custom message provides a correct message 1`] = `
"my custom message

   Line | Content
 >    1 | language: eno
"
`;

exports[`validation.excessName with a custom message provides correct selection metadata 1`] = `
Array [
  Array [
    0,
    0,
  ],
  Array [
    0,
    13,
  ],
]
`;

exports[`validation.excessName without a custom message provides a correct message 1`] = `
"An excess element named 'language' was found, is it possibly a typo?

   Line | Content
 >    1 | language: eno
"
`;

exports[`validation.excessName without a custom message provides correct selection metadata 1`] = `
Array [
  Array [
    0,
    0,
  ],
  Array [
    0,
    13,
  ],
]
`;
